app/src/test/kotlin/com/wire/android/config/CoroutineTestExtension.kt

```diff
@@ -0,0 +1,36 @@
+package com.wire.android.config
+
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.test.TestCoroutineDispatcher
+import kotlinx.coroutines.test.TestCoroutineScope
+import kotlinx.coroutines.test.resetMain
+import kotlinx.coroutines.test.setMain
+import org.junit.jupiter.api.extension.AfterEachCallback
+import org.junit.jupiter.api.extension.BeforeEachCallback
+import org.junit.jupiter.api.extension.ExtensionContext
+
+/**
+ * Add this JUnit 5 extension to your test class using
+ * @JvmField
+ * @RegisterExtension
+ * val coroutinesTestExtension = CoroutinesTestExtension()
+ *
+ * or:
+ *
+ * Annotating the class with
+ * @ExtendWith(CoroutineTestExtension::class)
+ */
+@ExperimentalCoroutinesApi
+class CoroutineTestExtension(
+    private val dispatcher: TestCoroutineDispatcher = TestCoroutineDispatcher()
+) : BeforeEachCallback, AfterEachCallback, TestCoroutineScope by TestCoroutineScope(dispatcher) {
+
+    override fun beforeEach(context: ExtensionContext?) {
```

Comment:
How is this context being used?

Reply:
In fact this it's not used (we don't need it) we are just overriding the `beforeEach`  and `afterEach` from the contracts `BeforeEachCallback` and `AfterEachCallback` interfaces, to set and reset the dispatcher to be used on testing context